core/trino-main/src/main/java/io/trino/metadata/MetadataListing.java

```diff
@@ -85,6 +86,20 @@ public static List<CatalogInfo> listCatalogs(Session session, Metadata metadata,
                 .collect(toImmutableList());
     }
 
+    public static Set<CatalogSchemaName> listAllAvailableSchemas(
```

Comment:
Is this method actually needed if all it does is delegating to `listCatalogNames` and `listSchemas`?